internal/pkg/mining/scheduler.go

```diff
@@ -51,56 +61,77 @@ func (s *timingScheduler) Start(miningCtx context.Context) (<-chan Output, *sync
 	// loop mining work
 	doneWg.Add(1)
 	go func() {
-		s.mineLoop(miningCtx, outCh, &doneWg)
+		err := s.mineLoop(miningCtx, outCh)
+		if err != nil {
+			outCh <- NewOutputErr(err)
+		}
 		doneWg.Done()
 	}()
 	s.isStarted = true
 
 	return outCh, &doneWg
 }
 
-func (s *timingScheduler) mineLoop(miningCtx context.Context, outCh chan Output, doneWg *sync.WaitGroup) {
-	// mineLoop is the main event loop for the timing scheduler.  It waits for
-	// a new epoch to start, polls the heaviest head, includes the correct number
+func (s *timingScheduler) mineLoop(ctx context.Context, outCh chan Output) error {
```

Comment:
Just FYI I am hoping to get rid of this channel and replace it with a synchronous block consuming function.